erasure-createfile.go

```diff
@@ -28,11 +28,11 @@ import (
 // erasureCreateFile - writes an entire stream by erasure coding to
 // all the disks, writes also calculate individual block's checksum
 // for future bit-rot protection.
-func erasureCreateFile(disks []StorageAPI, volume, path string, reader io.Reader, blockSize int64, dataBlocks int, parityBlocks int, writeQuorum int) (bytesWritten int64, checkSums []string, err error) {
+func erasureCreateFile(disks []StorageAPI, volume, path string, reader io.Reader, blockSize int64, dataBlocks int, parityBlocks int, algo string, writeQuorum int) (bytesWritten int64, checkSums []checkSumInfo, err error) {
 	// Allocated blockSized buffer for reading.
```

Comment:
it's correct to pass algo as a string instead of hard coding.

but code is more readable if erasureCreateFile returns `checkSums []string` and constructing checkSumInfo happens at one place (xl.PutObject) than filling some fields in erasureCreateFile and some fields in xl.PutObject